tatanka/spec/meshdex.md

```diff
@@ -0,0 +1,62 @@
+# DEX on Tatanka Mesh
+
+Implementing a peer-to-peer decentralized exchange on Tatanka Mesh will require
+substantial re-thinking of many existing DCRDEX protocol features. The primary
+difference between DCRDEX and DEX on Tatanka Mesh is that Tatanka Mesh clients
+take over the roll of maintaining order books.
+
+### Protocol Basics
+
+1) There are no pre-defined markets. A market is simply a broadcast channel,
+which is a topic-subject pair, e.g. topic = "market", subject = "dcr_btc". Any
+user can create a new market by simply subscribing to a non-existent channel.
+This means we need to eliminate most market configuration.
+    - Lot size will not be a constant. Instead, it will be chosen by the user
+    based on their preferences for limiting fee exposure. For example, the user
+    may set their fee exposure limit to 1%. This will hide any order (in their
+    local view of the order book) which have a lot size that might result in fee
+    losses > 1%. Only orders with compatible lot sizes can match. To ensure
+    orders fill completely, I propose that lot sizes must be a power of 2,
+    though other schemes are possibly conceivable.
+    - Max fee rates are similarly set by the user as part of the order, and
+    orders whose max fee rate falls below the fee rates provided by the mesh
+    oracle feed will be ignored or unbooked.
+    - If the mesh provides a fiat exchange rate feed, we can define the rate
+    step and parcel size as a exchange-wide constant in terms of USD, and these
+    values will be dynamically re-calculated for each market as rates change.
+    - We could potentially do away with epoch-based matching, but we may want to
+    keep it around, albeit in a different form. Our goal should be that clients
+    are reacting appropriately by creating matches for valid offers. Clients
+    can't just ignore offers based on their own arbitrary criteria. Ideally
+    they would accept the first offers that are sent to them, though this would
+    be very hard to enforce.
+    - I've considered doing away with order funding validation. The only purpose
+    of funding validation is to prevent empty wallets from spamming the market
+    with orders they have no intention to fulfill. This would of course result
+    in an account suspension, so such spamming cannot go on indefinitely. The
+    bonding system also requires users to lock up funds before using the
+    network, so there is still a barrier to such malicious behavior, though this
+    is not as strong a filter as funding validation would provide. It's
+    important to remember that the global worst-case scenario for
+    atomic-swap-based exchange is a refund, not lost funds, so a malicious
```

Comment:
> It's important to remember that the global worst-case scenario for atomic-swap-based exchange is a refund, not lost funds

Broadcasting a swap transaction and associated refund transaction still leads to **some** loss of funds in the form of transaction fees. While this loss is negligible on many chains, that isn't always true (especially if it happens repeatedly). It's a detail which is worth calling out IMO.